### ISSUE TITLE
deadlock after failed transfer

### ISSUE THREAD

**Issue by dev-random** [LuaLanes/lanes#135]
```lua
local lanes = require('lanes').configure()
local linda = lanes.linda()
lanes.gen('*', function()
	print(pcall(linda.receive, linda, 'tmp'))
end)()

-- just something to make send() succeed and receive() fail:
local payload = { lanes.require('socket').connect }

linda:send('tmp', payload)
print('we reach here and then deadlock')
lanes.sleep(2)
print('we never reach here')
```

**Comment by sonoro1234** [comment]
I am working with older lanes version but my result is:
```
we reach here and then deadlockfalse    Keeper #1: function 'socket/connect' not found in Lane #0x003488b8 des
tination transfer database.

we never reach here
```
I dont know why you are not getting this.

**Comment by benoit-germain** [comment]
HEAD version does seem to freeze as of Nov 8th, I'll have to check.
```
Lua 5.3.5  Copyright (C) 1994-2018 Lua.org, PUC-Rio
> dofile "deadlock.lua"
we reach here and then deadlock
false   Keeper #1: function 'socket/connect' not found in Lane #0000020BCCF8C5A8 destination transfer database.
```

After some investigation I see what's going on. The only way to fix it for good it to make all linda operations capture errors with an additional ```lua_pcall```. This will have a performance impact on programs that send heavy streams of data through lindas though.

**Comment by sonoro1234** [comment]
The same version LuaJIT 32 bits mingw
no deadlock
```
we reach here and then deadlockfalse
Keeper #1: function 'socket/connect' not found in Lane #0x0036e5c8 destination transfer database.
we never reach here
```

**Comment by benoit-germain** [comment]
Fixed by 0cc1c9c9dcea5955f7dab921d9a2fff78c4e1729.
